flink-core/src/test/java/org/apache/flink/core/memory/OffHeapUnsafeMemorySegmentTest.java

```diff
@@ -58,4 +61,31 @@ public void testCallCleanerOnFree() {
                 .free();
         assertTrue(cleanerFuture.isDone());
     }
+
+    @Test
+    public void testCallCleanerOnceOnConcurrentFree() throws InterruptedException {
+        final AtomicInteger counter = new AtomicInteger(0);
```

Comment:
I think we should also test throwing path, in that path the final `counter` should also be one.

Reply:
By "throwing path", do you mean the path that `checkMultipleFree` is activated? I think that should be covered by the current test case as long as the env is set, which I've verified manually.

Not sure if we need two different test cases with `checkMultipleFree` set differently. The two cases would have to be executed in different processes, because `checkMultipleFree` is determined only when the class `MemorySegment` is first loaded.

Reply:
Ideally, it would be better to test all paths in codes. But I also could not give a less intrusive solution to this. So, I am just fine with current situation.